datastore.go

```diff
@@ -443,13 +511,22 @@ func (b *batch) Put(ctx context.Context, key ds.Key, value []byte) error {
 	if b.ds.closed {
 		return ErrClosed
 	}
+
+	if b.ds.ttl > 0 {
+		return b.putWithTTL(key, value, b.ds.ttl)
+	}
+
 	return b.put(key, value)
 }
 
 func (b *batch) put(key ds.Key, value []byte) error {
 	return b.writeBatch.Set(key.Bytes(), value)
 }
 
+func (b *batch) putWithTTL(key ds.Key, value []byte, ttl time.Duration) error {
```

Comment:
IMO we should have a publicly exported version of this (e.g. as we have for `Datastore` and `txn`). Having this would enable you to just make a few wrapper structs that make it so you have a default TTL.

We can still have a default TTL for convenience, but it shouldn't be necessary.

Reply:
I wanted to do that as well, but `Batch` method returns a `ds.Batch`, and that interface only defines a `Put` method. So the caller would have to typecast to an unexported struct.